pkg/endpoint/policy.go

```diff
@@ -709,3 +710,47 @@ func (e *Endpoint) GetCIDRPrefixLengths() (s6, s4 []int) {
 	}
 	return e.desiredPolicy.CIDRPolicy.ToBPFData()
 }
+
+// UpdateVisibilityPolicy updates the visibility policy of this endpoint to
+// reflect the state stored in the proxy visibility annotation, if present,
+// in anno. If no such annotation is present, then the VisibilityPolicy for the
+// Endpoint will be empty, and will have no effect. If the proxy visibility
+// annotation cannot be parsed, an empty visibility policy is assigned to the
+// Endpoint.
+func (e *Endpoint) UpdateVisibilityPolicy(anno map[string]string) {
+	if err := e.lockAlive(); err != nil {
+		// If the endpoint is being deleted, we don't need to update its
+		// visibility policy.
+		return
+	}
+	defer e.unlock()
+
+	var (
+		nvp *policy.VisibilityPolicy
+		err error
+	)
+
+	if anno != nil {
```

Comment:
no need for this check

Reply:
The function is exported / public, so any caller could provide a nil map.

Reply:
so? line 743 won't panic